File: app/src/main/java/com/nimtego/plectrum/data/cache/Cache.java
package com.nimtego.plectrum.data.cache;

import com.nimtego.plectrum.data.entity.Album;

import io.reactivex.Observable;

public interface Cache {

    Observable<Album> get(final String name);

    void put(Album album);

    boolean isCached(final int userId);

    boolean isExpired();

    void evictAll();
}

File: app/src/main/java/com/nimtego/plectrum/data/cache/Cache.kt
package com.nimtego.plectrum.data.cache

import io.reactivex.Completable
import io.reactivex.Observable

interface Cache<Key : Any, Value : Any> {
    // replacement Deferred coroutines?
    // Deferred<Value?>, Deferred<Unit>
    fun get(key: Key): Observable<Value>
    fun put(key: Key, value: Value): Completable
    fun evict(key: Key): Completable
    fun evictAll(): Completable
}